rewrite-java/src/main/java/org/openrewrite/java/CommentImport.java

```diff
@@ -0,0 +1,96 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import lombok.With;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Option;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.Comment;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.TextComment;
+import org.openrewrite.marker.Marker;
+import org.openrewrite.marker.Markers;
+
+import java.util.UUID;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class CommentImport extends Recipe {
+
+    @Option(displayName = "Fully-qualified type name",
+            description = "Fully-qualified class name of the type.",
+            example = "org.junit.Assume")
+    String fullyQualifiedTypeName;
+
+    @Option(displayName = "Comment text",
+            description = "The text to add as a comment.",
+            example = "This type was removed.")
+    String commentText;
+
+    @Option(displayName = "Multiline comment",
+            description = "Add the comment as a multiline comment. Defaults to `false`.",
+            required = false)
+    @Nullable
+    Boolean multiline;
+
+    @Override
+    public String getDisplayName() {
+        return "Comments an import statement";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Comments an import statement.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.check(
+                new UsesType<>(fullyQualifiedTypeName, false),
+                new JavaVisitor<ExecutionContext>() {
+                    @Override
+                    public J visitImport(J.Import import_, ExecutionContext ctx) {
+                        import_ = (J.Import) super.visitImport(import_, ctx);
+                        if (import_.getMarkers().findFirst(ImportAlreadyCommented.class).isPresent()) {
+                            return import_;
+                        }
+                        if (import_.getTypeName().equals(fullyQualifiedTypeName)) {
+                            boolean multilineComment = Boolean.TRUE.equals(multiline);
+                            String formattedComment = multilineComment ? "\n" + commentText + "\n" : commentText;
+                            Comment comment = new TextComment(multilineComment, formattedComment, import_.getPrefix().getWhitespace(), Markers.EMPTY);
+                            return import_.withComments(ListUtils.concat(import_.getComments(), comment))
+                                    .withMarkers(import_.getMarkers().add(new ImportAlreadyCommented(UUID.randomUUID())));
```

Comment:
Looking at this now it appears you'd want to comment out the original import, instead of adding comment to the import; is that correct? In that case I think you'd have to remove this import, find whatever is the next LST element, and merge any comments on this import, that next LST element and a commented out import there.

Slightly more involved; but we'd used similar logic when we for instance add or remove an import: that also needs to shuffle around any existing comments and tag them onto the intuitively right LST element.